components/p3a/brave_p3a_star.cc

```diff
@@ -0,0 +1,144 @@
+/* Copyright 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/p3a/brave_p3a_star.h"
+
+#include <algorithm>
+#include <utility>
+#include <vector>
+
+#include "base/base64.h"
+#include "base/bind.h"
+#include "base/logging.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/string_split.h"
+#include "brave/components/p3a/brave_p3a_config.h"
+#include "brave/components/p3a/p3a_message.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service.h"
+#include "services/network/public/cpp/resource_request.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+
+namespace brave {
+
+namespace {
+
+constexpr std::size_t kP3AStarCurrentThreshold = 50;
```

Comment:
An interesting idea. My understanding is that _k_ must be universal across all clients. Unsure if STAR would support a dynamic _k_ value (cc @alxdavids ). It would also require some significant changes for the aggregator

Reply:
Happy to leave this one as an open issue for discussion. I'll drop a message in the STAR channel for now and if we feel it warrants changes we can open an issue later. Resolving this

Reply:
@kdenhartog that would be interesting to think about, I think it could work but I think it may be worth opening an issue against [sta-rs](https://github.com/brave/sta-rs/) to track the discussion.